docs/source/upgrade/index.rst

```diff
@@ -82,13 +83,103 @@ has one *nonconsuming* choice that accepts a *Coin* contract, archives
 this *Coin* contract and creates a *CoinWithAmount* contract with
 *amount* set to 1.
 
-.. literalinclude:: daml/Upgrade.daml
+.. literalinclude:: example/coinV2/daml/UpgradeFromCoinV1.daml
   :language: daml
   :start-after: -- UPGRADE_AGREEMENT_BEGIN
   :end-before: -- UPGRADE_AGREEMENT_END
 
-.. TODO Turn this into a full example project with multiple packages
-.. and explain builds and deployment.
 
-.. Building and deploying
-.. ======================
+Building and deploying version-1
+================================
+
+Let's see everything in action by following the steps to build and deploy version-1 of our coin. After this we'll see how to deploy and upgrade to version-2.
+
+First we'll need a sandbox ledger to which we can deploy.
+
+.. code-block:: none
+
+   $ daml sandbox --port 7001
+
+The version-1 project is named ``coinV1``. It contains just the original *Coin* definition template ``Coin``. Here is the ``daml.yaml`` setup
+
+.. literalinclude:: example/coinV1/daml.yaml
+  :language: yaml
+  :start-after: # BEGIN
+  :end-before: # END
+
+To make the example complete, we need a way to create some coins which have different parties for the issuer and owner. We take the standard approach of adding a :ref:`Propose-Accept workflow <intro propose accept>` to our example.
+
+.. literalinclude:: example/coinV1/daml/CoinV1.daml
+  :language: daml
+  :start-after: -- COIN_TRANSFER_BEGIN
+  :end-before: -- COIN_TRANSFER_END
+
+Now we can build and deploy version-1 of our Coin.
+
+.. code-block:: none
+
+   $ cd example/coinV1
+   $ daml build
+   $ daml deploy --port 7001
+
+
+Create some version-1 coins
+================================
+
+Let's create some coins. We'll use the navigator to connect to the ledger, and create two coins issued by Alice, and owned by Bob.
+
+.. code-block:: none
+
+   $ cd example/coinV1
+   $ daml navigator server localhost 7001
+
+We point a browser to ``localhost:4000``, and follow the steps:
```

Comment:
```suggestion
We point a browser to http://localhost:4000, and follow the steps:
```

Sphinx automatically detects links so this will be clickable.